nemo/export/trt_llm/nemo_ckpt_loader/nemo_file.py

```diff
@@ -70,69 +70,68 @@ def __init__(self, path: Union[Path, TarPath]) -> None:
 
 
-def get_extra_state_key(state_dict: dict) -> Optional[str]:
-    for key in state_dict.keys():
-        if '_extra_state/' in key:
-            return key
-    return None
+# Used only for the old, non-mcore path
+def standarize_distributed_scaling_factors(state_dict: dict) -> dict:
+    scales_dict = {k: v for k, v in state_dict.items() if 'extra_state' in k}
+    state_dict = {k: v for k, v in state_dict.items() if 'extra_state' not in k}
 
+    scales = {}
 
-def unpack_extra_state_key(key: str) -> Tuple[str, int]:
-    basename = key.split('/')[0]
-    size = int(key.split('/')[1].split('_')[-1])
-    return basename, size
+    for k, v in scales_dict.items():
+        v.seek(0)
+        extra_state = torch.load(v)
+        if extra_state is not None and 'scale_fwd' in extra_state:
+            scales[k + '.scale_fwd'] = extra_state['scale_fwd'].cpu()
 
+    combined_scales = {}
+    for k in scales:
+        if 'model.decoder.layers.' not in k:
+            continue
 
-def clear_loaded_extra_states(state_dict: dict, basename: str) -> dict:
-    """The scaling factors are originally saved to state_dict under the keynames 'basename/*'
-    The standardized representation is saved to 'basename.*'. This function clears the former from the state.
-    """
-    to_remove = [k for k in state_dict.keys() if basename + '/' in k]
-    for key in to_remove:
-        state_dict.pop(key)
-    return state_dict
+        decomposed = k.split('.')
+        combined = []
+        id = 0
+        decomposed[3] = str(id)
+        while '.'.join(decomposed) in scales:
+            combined.append(scales['.'.join(decomposed)])
+            id += 1
+            decomposed[3] = str(id)
 
+        del decomposed[3]
+        combined_scales['.'.join(decomposed)] = torch.stack(combined)
 
-def retrieve_scale(bytes: BytesIO) -> Optional[torch.Tensor]:
-    bytes.seek(0)
-    extra_state = torch.load(bytes)
-    if not extra_state or 'scale_fwd' not in extra_state:
-        return None
-    return extra_state['scale_fwd'].cpu()
+    return state_dict | combined_scales
 
 
-def load_scales_from_bytes(bytes_list: List[BytesIO]) -> Optional[torch.Tensor]:
-    scales = []
-    for bytes in bytes_list:
-        scale = retrieve_scale(bytes)
-        if scale is None:
-            return None
-        scales.append(scale)
-    return torch.stack(scales)
+def rename_extra_states(state_dict: dict) -> dict:
+    mcore_extra_states = {}
 
+    for key, value in state_dict.items():
+        if 'extra_state' not in key:
+            continue
 
-def load_scaling_factors(state_dict: dict, basename: str, size: int) -> Optional[torch.Tensor]:
-    keynames = [f'{basename}/shard_{layer}_{size}' for layer in range(size)]
-    bytes_list = [state_dict[keyname][0] for keyname in keynames]
-    return load_scales_from_bytes(bytes_list)
+        decomposed_sharded_key = key.split('/')
+        if not len(decomposed_sharded_key):
```

Comment:
Always `len(decomposed_sharded_key) >= 1` so this `if` is inactive?

Reply:
True, I've deleted this `if`